electron-app/src/websites/artconomy/artconomy.service.ts

```diff
@@ -0,0 +1,239 @@
+      status.username = authCheck.response.body.username;
+      status.loggedIn = true;
+    }
+    return status;
+  }
+
+  transformAccountData(data: ArtconomyAccountData) {
+    return { username: data.username, id: data.id, csrftoken: data.csrftoken, authtoken: data.authtoken };
+  }
+
+  getScalingOptions(file: FileRecord): ScalingOptions | undefined {
+    return { maxSize: FileSize.MBtoBytes(200) };
+  }
+
+  preparseDescription(text: string) {
+    text = UsernameParser.replaceText(text, 'fa', '[fa]$1[/fa]');
+    text = UsernameParser.replaceText(text, 'sf', '[sf]$1[/sf]');
+    text = UsernameParser.replaceText(text, 'da', '[da]$1[/da]');
+    text = UsernameParser.replaceText(text, 'ws', '[w]$1[/w]');
+    text = UsernameParser.replaceText(text, 'ib', '[iconname]$1[/iconname]');
+    return text;
+  }
+
+  parseDescription(text: string) {
+    text = super.parseDescription(text);
+    return text.replace(/\[hr\]/g, '-----');
+  }
+
+  private getRating(rating: SubmissionRating | string) {
+    switch (rating) {
+      case SubmissionRating.GENERAL:
+        return 0;
+      case SubmissionRating.MATURE:
+        return 1;
+      case SubmissionRating.ADULT:
+        return 2
+      case SubmissionRating.EXTREME:
+        return 3
+      default:
+        // Safest assumption.
+        return 2
+    }
+  }
+
+  checkAssetUpload(upload: HttpResponse<{id: string}>) {
+    if (!(upload.body.id)) {
+      return Promise.reject(
+          this.createPostResponse({
+            message: upload.response.statusMessage,
+            additionalInfo: JSON.stringify(upload.body),
+          }),
+      );
+    }
+  }
+
+  async postFileSubmission(
+    cancellationToken: CancellationToken,
+    data: FilePostData<ArtconomyOptions>,
+    accountData: ArtconomyAccountData,
+  ): Promise<PostResponse> {
+    const primaryAssetForm: any = {
+      'files[]': data.primary.file,
+    };
+
+    const thumbnailAssetForm = {
+      'files[]': data.thumbnail,
+    }
+    const requestOptions = { json: true, headers: {Authorization: `Token ${accountData.authtoken}`}};
+    this.checkCancelled(cancellationToken);
+    let upload = await Http.post<{ id: string }>(
+      `${this.BASE_URL}/api/lib/v1/asset/`,
+      undefined,
+      {
+        requestOptions,
+        type: 'multipart',
+        skipCookies: true,
+        data: primaryAssetForm,
+      },
+    );
+    let check = this.checkAssetUpload(upload)
+    if (check) {
+      return check
+    }
+
+    const primaryAsset = upload.body.id;
+    let thumbnailAsset: null|string = null
+    if (data.thumbnail) {
+      upload = await Http.post<{ id: string }>(
+          `${this.BASE_URL}/api/lib/v1/asset/`,
+          undefined,
+          {
+            requestOptions,
+            type: 'multipart',
+            skipCookies: true,
+            data: thumbnailAssetForm,
+          },
+      );
+      check = this.checkAssetUpload(upload)
```

Comment:
```check = await this.checkAssetUpload```

You can then remove the if (check) below as that will throw the rejection and be caught in the calling fn